packages/parcel-transformer/src/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { Transformer } from '@parcel/plugin';
+import compiledBabelPlugin from '@compiled/babel-plugin';
+import { parseAsync, transformFromAstAsync } from '@babel/core';
+import generate from '@babel/generator';
+
+/**
+ * Compiled parcel transformer.
+ */
+export default new Transformer({
+  canReuseAST() {
+    // Compiled should run before any other JS transformer.
+    return false;
+  },
+
+  async parse({ asset }) {
+    const code = await asset.getCode();
+    if (!asset.isSource || code.indexOf('@compiled/react') === -1) {
+      // We only want to parse files that are actually using Compiled.
+      // For everything else we bail out.
+      return undefined;
+    }
+
+    const ast = await parseAsync(code, {
+      filename: asset.filePath,
+      caller: { name: 'compiled' },
+    });
+
+    return ast;
+  },
+
+  async transform({ asset, ast }) {
```

Comment:
The AST isn't passed in here, you need to call `const ast = await asset.getAST();`

Reply:
according to https://v2.parceljs.org/plugin-system/transformer/ it is? 🤔

Reply:
I'll update that. The source of truth is this section: https://v2.parceljs.org/plugin-system/transformer/#Transformer

Reply:
No worries thanks, will update the code.

(As a side - is there any TypeScript types lying around for the `Transformer` class?)